### PR TITLE
Give a helpful diagnostic when the next struct field has an attribute

### LINKED ISSUE #100461: Attributes following a missing trailing comma produce over-eager recovery
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8a65ac81b182494a4856b59164bffbc2

```rust
struct Feelings {
    owo: bool
    #[allow(unused)] // If commented-out, we recover differently and do not get the additional error about no field `owo`
    uwu: bool,
}

impl Feelings {
    fn hmm(&self) -> bool {
        self.owo
    }
}
```

Having an attribute after the field with missing `,` turns the error from this (which is good):
```
   Compiling playground v0.0.1 (/playground)
error: expected `,`, or `}`, found `uwu`
 --> src/main.rs:2:14
  |
2 |     owo: bool
  |              ^ help: try adding a comma: `,`

error: could not compile `playground` due to previous error
```
To this, which contains unhelpful errors and possibly a huge number of them on a larger codebase:
```
Compiling playground v0.0.1 (/playground)
error: expected `,`, or `}`, found `#`
 --> src/main.rs:2:14
  |
2 |     owo: bool
  |              ^

error[E0609]: no field `owo` on type `&Feelings`
 --> src/main.rs:9:14
  |
9 |         self.owo
  |              ^^^ unknown field

For more information about this error, try `rustc --explain E0609`.
error: could not compile `playground` due to 2 previous errors
```

### PR DESCRIPTION
**Pull request by chenyukang**
Fixes #100461

### CODE REVIEW

File: compiler/rustc_parse/src/parser/item.rs [hunk 1]
```diff
@@ -1539,7 +1539,7 @@ impl<'a> Parser<'a> {
                     }
                 }
 
-                if self.token.is_ident() {
+                if self.token.is_ident() || self.token.kind == TokenKind::Pound {
```
Comment (compiler-errors) [inline compiler/rustc_parse/src/parser/item.rs:1542]: Should we look ahead for more than 1 token, i.e. check for both `# [` tokens? 

Also, please leave a comment explaining why we're checking for `TokenKind::Pound`

**Comment by compiler-errors** [comment]
@chenyukang you need to write "fix" or "fixes" in the description to link it to an issue, "fixing" doesn't work :sweat_smile:

**Comment by chenyukang** [comment]
> @chenyukang you need to write "fix" or "fixes" in the description to link it to an issue, "fixing" doesn't work 😅 I edited it.

Thanks!

File: compiler/rustc_parse/src/parser/item.rs (hunk 1 above)
Comment (chenyukang) [reply compiler/rustc_parse/src/parser/item.rs]: Fixed!

**Comment by fee1-dead** [comment]
LGTM! Thanks

@bors r+ rollup

**Pull request merged**